crates/storage_impl/src/kv_router_store.rs

```diff
@@ -269,7 +269,7 @@ impl<T: DatabaseStore> KVRouterStore<T> {
                 key_store.merchant_id.clone().into(),
             )
             .await
-            .change_context(errors::StorageError::DecryptionError)
+            .map_err(|error| error_stack::report!(errors::StorageError::from(error)))
```

Comment:
And because there's a possibility of losing context about the original error, I'd prefer `change_context()` being called instead.